### ISSUE TITLE
Issue with Bibliocraft Containers and the Storage Scanner

### ISSUE THREAD

**Issue by Tanguygab** [McJtyMods/RFTools#1886]
Issue description: 
I can't craft Items from the Storage Scanner if they aren't in my inventory when the it scan containers from Bibliocraft.

Steps to reproduce: 
Scan a container from Bibliocraft (Tool Rack,Shelf,..)

Versions:
* Minecraft: 1.12.2
* Forge: 14.23.5.2772
* Bibliocraft: 2.4.5
* McJtyLib: 3.0.6
* RFTools: 7.59

Possibly incompatible mods and versions: 
Bibliocraft 2.4.5

Relevant logs, if any: 
the log when I click on the button to craft items
[bug.txt](https://github.com/McJtyMods/RFTools/files/2658255/bug.txt)

**Comment by mhelvens** [comment]
@josephcsible @Tanguygab 

This seems to still be a problem (Bibliocraft 2.4.5 + RFTools 7.72).

It does look like Bibliocraft fails to override the abstract `getStackInSlot` method. So yes, they seem to be at fault. But perhaps RFTools should handle the exception more gracefully?

I suggest that you simply catch the exception in [your own `getStackInSlot` method](https://github.com/McJtyMods/RFTools/blob/bc0d0360e673aed0179f160b4fd266f9770b6c04/src/main/java/mcjty/rftools/craftinggrid/TileEntityItemSource.java#L35) and log it as a warning, instead of allowing it to propagate further up the call stack. Something like this:

```Java
private static ItemStack getStackInSlot(Object inv, int slot) {
	try {
		if (inv instanceof IItemHandler) {
		    return ((IItemHandler) inv).getStackInSlot(slot);
		} else if (inv instanceof IInventory) {
		    return ((IInventory) inv).getStackInSlot(slot);
		}
	} catch (AbstractMethodError e) {
		Logger.warn(e.getMessage());
	}
	return ItemStack.EMPTY;
}
```

That way, rather than misbehave in the mere presence of a bookcase, the crafting grid would just fail to detect any items stored in one, but otherwise function properly. (Or perhaps someone more familiar with these APIs could even write a backup method for getting item stacks from older mods?)

I'm afraid I can't write a proper PR. I've never worked on Minecraft mods, so I'm not set up to test this properly. But it should be just this simple, shouldn't it?

**Comment by josephcsible** [comment]
You want us to catch an `Error` (not even an `Exception`) for the purpose of working around a bug in another mod? That's an absolutely terrible coding practice, and no, there's not a way of working around it to do the right thing. Did anyone ever open a report about this in Bibliocraft's tracker? The only right way to fix this is for them to do it.

**Comment by mhelvens** [comment]
@josephcsible 

I agree that it's usually bad practice. But isn't it acceptable when you don't have control over the code that's causing the problem? The alternative is buggy behavior in big modpacks like Enigmatica 2. And it's not obvious that Bibliocraft is the culprit, so it makes RFTools look bad.

And yes, Tanguygab [reported the bug](https://github.com/Nuchaz/BiblioCraft/issues/629) there, as you requested. In fact, this bug has been reported there [about nine times](https://github.com/Nuchaz/BiblioCraft/issues?q=is%3Aissue+is%3Aopen+getStackInSlot), and not once did they ever respond.

**Comment by Macleykun** [comment]
https://pastebin.com/Fdq4Aemh
Had this issue just now, saw this issue.
Using Mac Mega Modpack.

**Comment by Tanguygab** [comment]
I've just realized that the source code of the mod is now available https://github.com/Nuchaz/BiblioCraft-Source
So if you've got a dev or know how to code, you could try to fix it.

Personally, I don't use the mod anymore, well, not using 1.12.2 at all either, so it doesn't really interest me, nice thing that Nuchaz finally released the source code though =)

**Comment by Macleykun** [comment]
> I've just realized that the source code of the mod is now available https://github.com/Nuchaz/BiblioCraft-Source So if you've got a dev or know how to code, you could try to fix it.
> 
> Personally, I don't use the mod anymore, well, not using 1.12.2 at all either, so it doesn't really interest me, nice thing that Nuchaz finally released the source code though =)

It’s tottaly cool
I kinda feel indifferent about 1.12.2. Like I wanna move on but feel like I still miss some favs mods and all
Anyhow, only wanted to throw in my error :) for whoever stumbles upon this and wants to fix it
